### PR TITLE
add urb decoder, unit tests, and integration test with pcap reader & …

### LINKED ISSUE #18: feat: implement URB decoder (Control + Bulk transfers)
implement the URB decoder for Control and Bulk transfer types

## location
`bsu_tool/urb_decoder.py`

## behavior
- accept raw packet bytes from the `pcap-ng` reader
- decode the 64-byte USB header into a structured URB record
- extract key fields: `id`, type (submission/completion), `transfer_type`, `direction`, `bus_num`, `dev_num`, endpoint, status, length, data payload
- handle both Control & Bulk transfer types
- correctly pair submissions with their completions using the URB id field

## acceptance criteria
- [x] decodes Control transfers correctly against reference captures
- [x] decodes Bulk transfers correctly against reference captures
- [x] submissions and completions correctly paired by URB id
- [x] returns structured typed dataclass or TypedDict with no untyped dicts
- [x] full type annotations & passes pyright strict
- [x] ruff clean
- [x] tests written in `tests/unit/test_urb_decoder.py`
- [x] at least one test per transfer type

## notes
- USB 2.0 spec Chapter 8 is the normative reference for URB structure found [here](https://archive.org/download/USB-2.0/Universal%20Serial%20Bus%20Specification%2C%20Revision%202.0.pdf)
- Ask Claude to walk through the 64-byte USB header layout before starting & document how you used Claude
- Interrupt transfers are Milestone 2. do not implement yet

## dependencies
blocked by [*feat: implement pcap-ng reader*](https://github.com/bsu-tool/bsu-tool/issues/17)

### PR DESCRIPTION
**Pull request by AliciaCurths**
closes #18 urb decoder

## What does this PR do?

This pull request introduces a new USB Request Block (URB) decoder module, `urb_decoder.py`, which provides robust decoding and pairing of USB packets captured via Linux usbmon, along with comprehensive integration tests and additional real-capture unit tests. The main focus is on accurate decoding of control and bulk transfers, pairing submissions and completions, and validating the pipeline using sanitized real-world capture files.

The most important changes are:

**New URB Decoder Implementation:**

* Adds `bsu_tool/urb_decoder.py`, a module that decodes raw usbmon packet data into strongly-typed `UrbRecord` objects and pairs them into `UrbTransaction` objects. It supports both legacy (48-byte) and modern (64-byte) usbmon formats, handles control and bulk transfers, and provides clear error handling for malformed or unsupported transfers.

**Integration and Unit Testing with Real Captures:**

* Introduces `tests/int/test_urb_decode_goodix.py`, an integration test suite that runs the full pcapng → urb_decoder pipeline on a sanitized Goodix fingerprint sensor capture. It verifies correct decoding, field values, transaction pairing, and edge cases (such as orphans and monotonic timestamps). 

* Expands `tests/unit/test_pcapng_reader.py` to include tests that parse and validate the structure of the same sanitized Goodix capture, checking for correct block counts, option redactions, and interface details. 

**Test Data and Fixtures:**

* Adds tests/unit/test_urb_decoder.py, a unit test suite that exercises the decoder against hand-crafted usbmon headers — covering control and bulk transfers, both link-type formats, timestamp composition, direction decoding, error cases, and all pair_urbs edge cases (orphans, double submissions, error events).

* Adds fixtures and test setup to ensure the presence and correct parsing of the sanitized capture file, enabling reliable, reproducible integration and unit tests on real-world data. 

**What's Not Included:**

* Interrupt transfer types are Milestone 2 — urb_decoder.py currently raises UnsupportedTransferTypeError.

## How was it tested?

Ruff, Pyright, and Pytest

## Checklist

- [x] PR description includes `closes #N`
- [x] No unintended files included in this PR

### CODE REVIEW

File: bsu_tool/urb_decoder.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+
+    event_type = _decode_event_type(event_byte)
+    transfer_type = _decode_transfer_type(xfer_byte)
+
+    direction: Direction = "in" if epnum & _ENDPOINT_DIR_MASK else "out"
+    endpoint = epnum & _ENDPOINT_NUM_MASK
+
+    setup: bytes | None = setup_bytes if transfer_type == "control" and flag_setup == _FLAG_SETUP_PRESENT else None
+
+    data = packet_data[header_size : header_size + captured_length]
```
Comment (JH12333) [inline bsu_tool/urb_decoder.py]: At this point the usbmon header appears to be present and parseable, but I do not think we have yet confirmed that its declared `captured_length` fits within `len(packet_data) - header_size`. If a malformed packet reported a larger captured length than the bytes actually available, this slice might silently return truncated data instead of raising `MalformedUsbmonHeaderError`.

**Review by JH12333 — CHANGES_REQUESTED** [review]
Non-blocking: the real-capture integration coverage looks strong for bulk traffic. I may have missed another test, but relative to Issue #18’s acceptance criteria, it looks like control-transfer decoding is currently covered by unit tests rather than by a reference capture. That may be completely acceptable for this PR, but it might be worth double-checking whether the current coverage is sufficient or whether a control reference-capture case should be added later.

File: bsu_tool/urb_decoder.py (hunk 1 above)
Comment (AliciaCurths) [reply bsu_tool/urb_decoder.py]: Added validation of capture length against actual bytes and updated urb unit test to test it. New capture file includes enumeration and fingerprint enrollment (control and bulk transfer types) and the integration test was updated to use the new capture file.

**Review by JH12333 — APPROVED** [review]
